tests/components/heos/test_config_flow.py

```diff
@@ -190,3 +190,151 @@ async def test_reconfigure_cannot_connect_recovers(
     assert config_entry.unique_id == DOMAIN
     assert result["reason"] == "reconfigure_successful"
     assert result["type"] is FlowResultType.ABORT
+
+
+async def test_options_flow_signs_in(
+    hass: HomeAssistant, config_entry, controller
+) -> None:
+    """Test options flow signs-in with entered credentials."""
+    config_entry.add_to_hass(hass)
+
+    # Start the options flow. Entry has not current options.
```

Comment:
here I'd prefer parametrized tests, because then we can test that we can recover from each of the individual failures